distribution/lib/Standard/Table/0.0.0-dev/src/Data/Type/Value_Type.enso

```diff
@@ -0,0 +1,243 @@
+from Standard.Base import all
+import Standard.Base.Errors.Illegal_Argument.Illegal_Argument
+
+from project.Errors import Invalid_Value_Type
+
+## Type to represent the different sizes of integer or float possible within a database.
```

Comment:
Am I right this is the core part of the PR? At least the file name `Table/**/Value_Type.enso` matches the title "..Value Types in Table..".

Shouldn't there be an example of how the `Value_Type` is supposed to be used from an end user perspective? 
- I can envision some form of conversions from/to Enso types. 
- `TruffleObject` provides special support for converting of objects between systems/languages/types
- Wrapping the values into Enso `Value_Type` is unlikely the most effective way
- but I'd need to understand the use-case more...

Reply:
Yep, you are right about the file being important :)

As for examples, I guess more will be available once we implement functions like `cast`. For now you can see how it's used in [`Table_Spec`](https://github.com/enso-org/enso/blob/0fb07c66ebba4595ac9ee204b5bd1cad225a1f69/test/Table_Tests/src/In_Memory/Table_Spec.enso#L100-L111) or for example [`Postgres_Spec`](https://github.com/enso-org/enso/blob/0fb07c66ebba4595ac9ee204b5bd1cad225a1f69/test/Table_Tests/src/Database/Postgres_Spec.enso#L93-L157).

Currently we are not performing conversions to/from Enso types, at least not directly, the concepts are a bit parallel. We could do these, but there was no use-case yet.

As for TruffleObject and mappings - I'm not sure we need this either. Sounds interesting, maybe could be useful if we have a usecase.

Currently, the `Value_Type` is meant to provide metadata of what kind of values may be stored in a Table Column. It is tailored to work well with both in-memory and Database columns, so the type system is built around how SQL types tend to work.